### PR TITLE
[Custom Reports] Feature: introduce initial sort to sql definition for report - fixes issue 12288

### LINKED ISSUE #12288: [Improvement]: Add "order by" to custom report with sql adapter
### Improvement description

In custom reports with the SQL adapter there should be the possibility to specify an order by. The order should be done when initially loading the data and when exporting the data.
What do you think?

### PR DESCRIPTION
**Pull request by frithjof**
adds the ability to supply an inital Order By to Grid and chart in Custom Report

Should be sufficent for most cases that use Chart because chart does not update with changing sort in Data

### CODE REVIEW

**Comment by frithjof** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mcop1** [comment]
Hello @frithjof ,

thank you, looks good to me! The only thing we could probably do is to create a function in the controller for getting the order and direction out of the configuration. This would reduce code duplication. What do you think?

**Comment by frithjof** [comment]
yeah sure was also thinking about that but did not want to refactor too much to make review easier.

will refactor that part then

**Comment by frithjof** [comment]
is there a way to mark parts that have to be adjusted for studio ui?

**Comment by mcop1** [comment]
> is there a way to mark parts that have to be adjusted for studio ui?

As of yet there isn´t, unfortunately we will have to look at all the ui parts separately anyways.

File: bundles/CustomReportsBundle/src/Controller/Reports/CustomReportController.php [hunk 1]
```diff
@@ -510,4 +486,26 @@ public function isValidConfigName(string $configName): void
...
+
+    /*
+     * gets the sort, direction, filters, drilldownfilters from grid or initial config
+     */
+    protected function getSortAndFilters(Request $request, \stdClass $configuration): array
+    {
+        $sortingSettings = null;
+        $sort = null;
+        $dir = null;
+        if(class_exists('\Pimcore\Bundle\AdminBundle\Helper\QueryParams')) $sortingSettings = \Pimcore\Bundle\AdminBundle\Helper\QueryParams::extractSortingSettings(array_merge($request->request->all(), $request->query->all()));
```
Comment (mcop1) [inline bundles/CustomReportsBundle/src/Controller/Reports/CustomReportController.php]:
```suggestion
    private function getSortAndFilters(Request $request, \stdClass $configuration): array
    {
        $sortingSettings = null;
        $sort = null;
        $dir = null;
        if(class_exists('\Pimcore\Bundle\AdminBundle\Helper\QueryParams')) {
        $sortingSettings = \Pimcore\Bundle\AdminBundle\Helper\QueryParams::extractSortingSettings(array_merge($request->request->all(), $request->query->all()));
        }
```

Comment (mcop1) [reply bundles/CustomReportsBundle/src/Controller/Reports/CustomReportController.php]: We can make the method private, right? No need for it to be protected.
And SonarCloud is complaining about missing {} for the if statement, I tried to add it in the suggestion, but couldn´t get the formatting right in the browser, sorry.

**Comment by mcop1** [comment]
Looks good now, thank you for the quick replies/commits :100:

**Pull request merged**